troposphere/static/js/actions/instance/stop.js

```diff
@@ -1,58 +1,88 @@
+import Backbone from "backbone";
 
 import InstanceConstants from "constants/InstanceConstants";
+import Instance from "models/Instance";
 import InstanceState from "models/InstanceState";
-import Utils from "../Utils";
 import InstanceActionRequest from "models/InstanceActionRequest";
 
-export default {
+import Utils from "../Utils";
 
-    stop: function(params) {
-        if (!params.instance)
-            throw new Error("Missing instance");
-
-        var instance = params.instance,
-            instanceState = new InstanceState({
-                status_raw: "active - powering-off",
-                status: "active",
-                activity: "powering-off"
-            }),
-            originalState = instance.get("state"),
-            actionRequest = new InstanceActionRequest({
-                instance: instance
-            });
+/**
+ * Defines the `stop` steps for a single instance
+ */
+function handleOne(instance){
+    // if we've been passed a model via "selected resources" that
+    // is *not* an Instance, then just skip it - _move on_
+    if (!(instance instanceof Instance)) return;
```

Comment:
I've been pondering this, and I really stand behind the decision to not raise an exception here. 

I was hoping to make a _best_ effort to carry the community member requested action. An exception would halt the attempt. By skipping, you don't try to do something you know will not work. 

I do think that I am _tainting_ `map` by using it to drive side-effects (and, by skipping, I am breaking the conceptual expectation regarding the resulting cardinality). Maybe I should have used `forEach` to communicate that this is purely iteration.